### PR TITLE
Secret Handshake approach

### PR DESCRIPTION
**Pull request by bobahop**
Approach for Secret Handshake.

### CODE REVIEW

**Comment by bobahop** [comment]
@ErikSchierboom: C Secret Handshake approach ready for review.

File: exercises/practice/secret-handshake/.approaches/introduction.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Introduction
+
+There are various wys to solve Secret Handshake.
```
Comment (ErikSchierboom) [inline exercises/practice/secret-handshake/.approaches/introduction.md:3]:
```suggestion
There are various ways to solve Secret Handshake.
```

**Review by ErikSchierboom — COMMENTED** [review]
One tiny nit

File: exercises/practice/secret-handshake/.approaches/introduction.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+One approach is to iterate once even if the actions are meant to be reversed.
+
+## Approach: Iterate once
+
+**secret_handshake.h**
+```c
```
Comment (wolf99) [inline exercises/practice/secret-handshake/.approaches/introduction.md:9]: Need empty line before code field

File: exercises/practice/secret-handshake/.approaches/iterate-once/content.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Iterate once
+
+**secret_handshake.h**
+```c
```
Comment (wolf99) [inline exercises/practice/secret-handshake/.approaches/iterate-once/content.md:4]: Need empty line before code field

File: exercises/practice/secret-handshake/.approaches/iterate-once/content.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    if (n & 16) {signal = 3; signal_incr = -1; end = -1;}
+
+    for (; signal != end; signal+=signal_incr)
+        if (n & 1 << signal)
+            output[output_idx++] = signals[signal];
+    return output;
+}
+```
+
+This approach starts by defining an array of pointers to `const` `char`s.
```
Comment (wolf99) [inline exercises/practice/secret-handshake/.approaches/iterate-once/content.md:35]: Question: would it be more accurate to say " pointers to `const char' strings"?

Maybe it's a linguistic thing.
There is an array that has multiple pointers each one to a string of const chars.

**Review by wolf99 — DISMISSED** [review]
Great!
Some small nits only

File: exercises/practice/secret-handshake/.approaches/iterate-once/content.md (hunk 4 above)
Comment (bobahop) [reply exercises/practice/secret-handshake/.approaches/iterate-once/content.md]: Changed to

`This approach starts by defining an array of pointers to `const` `char` strings.`

**Comment by bobahop** [comment]
@ErikSchierboom and/or @wolf99 . Made all suggested changes, so should be ready for merge now.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**